onnxruntime/core/providers/cpu/tensor/gather_elements.cc

```diff
@@ -124,7 +125,12 @@ static std::vector<int64_t> parse_and_validate_indices_tensor(const Tensor* indi
 
   return indices_data;
 }
-
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#ifdef HAS_CLASS_MEMACCESS
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
```

Comment:
>ragma GCC diagnostic ignored "-Wclass-memaccess" [](start = 2, length = 48)

Can you give an example of the code it complains about?

Reply:
In the code
```cpp
 if (is_string) { 
          output_data[++output_counter] = input_data[base_offset + indices_data[++indices_counter]];        
        } else {
          memcpy(output_data, input_data + (base_offset + indices_data[++indices_counter]) * element_size, element_size);
          output_data += element_size;        
        }
```

GCC complains that the code is trying to use memcpy to copy strings, which has non-trivial copy constructor.

Reply:
https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=81703&view=logs&j=ad86a3d4-3ae2-514b-dcc2-78fb9656b17e